examples/spec/lcdproc/LCDd.ini

```diff
@@ -2308,17 +2298,16 @@ description = "Specifies the displaytype.
 type = boolean
 check/type = boolean
 default = no
-description = "'no' if display connected serial, 'yes' if connected parallel.
-I.e. serial by default"
+description = "'no' if display connected serial, 'yes' if connected parallel."
 
 [serialvfd/custom-characters]
 type = unsigned_long
 check/type = unsigned_long
 check/validation = ^([1-9]\d*|0)$
 check/validation/match = LINE
 check/validation/message = Must be any positive number including 0
```

Comment:
Why not use range instead of this validation?

@Piankero In the error concept, it would be nice to always have something like the "check/validation/message", so that spec writers can improve the error messages.

Reply:
> In the error concept, it would be nice to always have something like the "check/validation/message", so that spec writers can improve the error messages.

This is especially true, if the global plugin is used for command line options. We have to position it before all of the validation plugins, so the error message will be about some `proc` key being invalid. And the user would have to know which key corresponds to which command-line option.

Reply:
> it would be nice to always have something like the "check/validation/message", so that spec writers can improve the error messages.

What exactly do you mean? A metadata which is added to the parentkey which solves as the optional "solution" which I suggested in the concept?

Reply:
As I understood it, we introduce a metakey e.g. `info/error`. We then provide a way of specifying which key caused an error, and the name of this key and its `info/error` would be stored together with the error.

`info/error` would be given by application developers writing a specification, a "solution" can only be given by developers of Elektra and its plugins IMO. If the `check/path/mode` validation fails, a solution is e.g. "Ensure you have write access to file XYZ". This can't be known when writing the specification, but in most cases "solution" should be obvious from the error description.

`info/error` wouldn't really be useful for `path`. It is more useful in cases where the specification is complicated and needs an explanation. For example if regexes are involved, or if the `conditional` plugin is used, e.g. "If /some/key is set to XYZ, this must not be set to ABC".

`info/error` should be an explanation of what the specification is meant to enforce, while `solution`  describes the simplest way to get rid of the error.

---

Note: If we can specify in a standardized way which key caused an error, it would be nicer to store the command-line options of this key with error. Then it wouldn't be necessary to add `info/error` to every key with a command-line option.

Reply:
Thank you @kodebach for that input, I will consider it in the concept.

> Why not use range instead of this validation?

The range plugin has no possibility to just set a minimum value but no maximum. 
Theoretically I can delete the validation anyway and just let the type plugin check for it. What do you say about that?

Reply:
Did you try it? @tom-wa sometimes has undocumented features, maybe "0-" works. If it works, please document it.

Reply:
> Theoretically I can delete the validation anyway and just let the type plugin check for it. What do you say about that?

The type plugin might also not provide the best error message. Hand-written error messages in the specification will often outperform generic error messages. I hope I will read an analysis of this in your thesis.